### PR TITLE
Mashraf/tracking

### PR DESCRIPTION
**Pull request by 22mahmoud**

### CODE REVIEW

File: src/api/profile.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import Cookies from 'js-cookie';
+import humps from 'humps';
+
+import config from '../../config';
+
+const API_URL: string = config('apiURL');
+
+const SESSION = `${API_URL}/v1/profile/session/`;
```
Comment (Ananasship) [inline src/api/profile.ts]: nit: we can rename this to `SESSION_URL`

File: src/api/profile.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+const RECITED_AYAHS = `${API_URL}/v1/profile/recited_ayahs/`;
```
Comment (Ananasship) [inline src/api/profile.ts]: Same here, `RECITED_AYAHS_URL`

File: src/api/profile.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  surahs: ISessionSurah[];
+}
+
+export interface ISessionData {
+  count: number;
+  results: ISession[];
+}
+
+interface IRecitedAyahsBody {
+	surah: number;
```
Comment (Ananasship) [inline src/api/profile.ts]: nit: spacing

File: src/api/profile.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+export const sumbitRecitedAyah = async (body: IRecitedAyahsBody) => {
+  try {
+    const res = await fetch(RECITED_AYAHS, {
+      method: 'POST',
+      credentials: 'include',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Token ${Cookies.get('authToken')!}`,
```
Comment (Ananasship) [inline src/api/profile.ts]: What does the `!` at the end do?

File: src/api/profile.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
+import Cookies from 'js-cookie';
```
Comment (Ananasship) [inline src/api/profile.ts]: should this file be named `Profile` or is there a better name for it?

File: src/pages/Progress/index.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { withCookies } from 'react-cookie';
+import _ from 'lodash';
+import format from 'date-fns/format';
+import getHours from 'date-fns/getHours';
+import Helmet from 'react-helmet';
+
+import { Containr, Body, Title } from './styles';
```
Comment (Ananasship) [inline src/pages/Progress/index.tsx]: typo: container

File: src/pages/Progress/index.tsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  render() {
+    const { loading, data } = this.state;
+    if (loading) {
+      return null;
+    }
+
+    return (
+      <Containr>
+        <Helmet>
+          <title>Weekly Progress</title>
```
Comment (Ananasship) [inline src/pages/Progress/index.tsx]: Why is this under `<Helmet>`?

File: src/pages/Transcribe/index.tsx [hunk 8]
```diff
@@ -197,6 +199,9 @@ class Transcribe extends React.Component<IProps, IState> {
   };
 
   handleStopRecording = async () => {
+    if (this.props.isAuthenticated && this.state.previousAyahs.length) {
+      console.log(this.state.previousAyahs);
```
Comment (Ananasship) [inline src/pages/Transcribe/index.tsx]: Are we currently making an API call when we stop recording? Should we add a TODO saying that we're working on changing this

Comment (22mahmoud) [reply src/pages/Transcribe/index.tsx]: this for old debug I'll delete it
we actually now use the new api

File: src/api/profile.ts (hunk 4 above)
Comment (22mahmoud) [reply src/api/profile.ts]: It is a way to tell the compiler "this expression cannot be null or undefined here, so don't complain about the possibility of it being null or undefined." 

[https://stackoverflow.com/a/42274019](https://stackoverflow.com/a/42274019)

**Pull request merged**